src/modules/homeView.js

```diff
@@ -0,0 +1,144 @@
+import { popupView } from './commentPopup.js';
+
+const homeContainer = document.querySelector('#home-container');
+
+// render data of the homepage
+const homepageView = async (data) => {
+  const rocketsContainer = document.querySelector('.rockets-container');
+  let rockets = '';
+
+  try {
+    const likesData = await getLikes(); // Retrieve the likes data from the API
+    const likesMap = new Map();
+
+    /* eslint-disable camelcase */
+    likesData.forEach(({ item_id, likes }) => {
+      // Create a map of rocketId to likes count
+      likesMap.set(item_id, likes);
+    });
+
+    data.forEach((rocket) => {
+      /* eslint-disable camelcase */
+      const rocketId = rocket.rocket_id;
+      const rocketName = rocket.rocket_name;
+      const rocketImage = rocket.flickr_images;
+      const successRate = rocket?.success_rate_pct;
+      const costPerLaunch = rocket?.cost_per_launch;
+      const weightMass = rocket?.mass.kg;
+      const rocketType = rocket?.rocket_type;
+
+      // Get the likes count from the map, defaulting to 0 if not found
+      const rocketLikes = likesMap.get(rocketId) || 0;
+      // the rockets views 🚀🚀
+      rockets += `
+      <li class="each-rocket" id="${rocketId}}">
+        <img src="${rocketImage}" alt="" class="list-item-image">
+        <section class="item">
+          <h3 class="rockec-name">${rocketName}</h3>
+          <section class="like-and-icon">
+          <i class="fa fa-heart" data-rocket-id="${rocketId}""></i>
+          <div class="like-count" id="likeCount${rocketId}">${rocketLikes} likes</div>
+          </section>
+        </section>
+        
+        <section class="button-container">
+        <button 
+          type="button" 
+          class="comments-button" 
+          data-rocket-id="${rocketId}" 
+          data-rocket-image="${rocketImage}"
+          data-rocket-name="${rocketName}"
+          data-rocket-type=${rocketType}
+          data-rocket-rate=${successRate}
+          data-rocket-cost=${costPerLaunch}
+          data-rocket-weight=${weightMass}
+        >
+        comments
+        </button>
+        </section>
+      </li>
+      `;
+    });
+    rocketsContainer.innerHTML = rockets;
+
+    // Add event listener to toggle heart icon color and track the icon button clicked
+    const likeIcons = document.querySelectorAll('.like-and-icon i');
+    likeIcons.forEach((icon) => {
+      icon.addEventListener('click', async () => {
+        const rocketId = icon.getAttribute('data-rocket-id');
+        icon.classList.toggle('liked');
+
+        try {
+          // initialize and add some likes to the likes count👍
+          await likeRocket(rocketId);
+        } catch (error) {
+          /* eslint-disable no-console */
+          console.log(error.message);
+          /* eslint-enable no-console */
+        }
+
+        // call the endpoint get the likes👍👍
+        try {
+          const data = await getLikes();
+          /* eslint-disable camelcase */
```

Comment:
- [ ] Please avoid disabling linter rules as they serve as a way to provide a consistent code format and ensure you use best practices. Instead, consider fixing the errors! 🤓 🔧

Reply:
resolved